### PR TITLE
Use clippy as a linter Close #29

### LINKED ISSUE #29: Use clippy as linter for Rust code
### Description

Project will be using [Clippy](https://github.com/rust-lang/rust-clippy) as linter for Rust code.

- _clippy.toml_ file needs to be created with following attributes:

```
# cyclomatic complexity is not always useful
cognitive-complexity-threshold = 100
# types are used for safety encoding
type-complexity-threshold = 10000
# we won’t allow too many function arguments
too-many-arguments-threshold = 13
```

- GitHub action needs to be created to check _Clippy_ warnings when a PR is created and when a PR is merged

### Acceptance Criteria

- _clippy.toml_ file exists in project's root directory
- GitHub action is created

### PR DESCRIPTION
**Pull request by hrmhatef**
### What was the problem?

This PR resolves https://github.com/LiskHQ/lisk-db/issues/29

### How was it solved?

- Remove all clippy errors as linter
- Update CI to run the clippy command

### How was it tested?

All test passed.

### CODE REVIEW

**Review by matjazv — APPROVED** [review]

File: src/smt_db.rs [hunk 1]
```diff
@@ -3,22 +3,21 @@ use std::collections::HashMap;
 use crate::consts;
 use crate::smt;
 
-pub struct SMTDB<'a> {
-    db: Box<&'a rocksdb::DB>,
+pub struct SmtDB<'a> {
```
Comment (shuse2) [inline src/smt_db.rs:6]: We usually use capital for abbreviated terms (ex: HTTP). Do we need to change this?

**Review by shuse2 — CHANGES_REQUESTED** [review]

File: src/smt_db.rs (hunk 1 above)
Comment (hrmhatef) [reply src/smt_db.rs:6]: Yes, the linter `clippy` suggested that changes based on the following error https://rust-lang.github.io/rust-clippy/master/index.html#upper_case_acronyms.

File: src/state_db.rs [hunk 2]
```diff
@@ -659,7 +681,8 @@ impl StateDB {
         let channel = ctx.channel();
 
         thread::spawn(move || {
-            let result = smt::SMT::verify(&parsed_query_keys, &proof, &state_root, key_lengh);
+            let result =
+                smt::SparseMerkleTree::verify(&parsed_query_keys, &proof, &state_root, key_lengh);
```
Comment (matjazv) [inline src/state_db.rs:685]: Spelling typo.

```suggestion
                smt::SparseMerkleTree::verify(&parsed_query_keys, &proof, &state_root, key_length);
```

**Review by matjazv — CHANGES_REQUESTED** [review]

File: package.json [hunk 3]
```diff
@@ -31,7 +31,8 @@
     "clean": "cargo clean && rm -rf bin-package && rm -rf build",
     "test:node": "jest",
     "test:rust": "cargo test",
-    "rust:fmt": "cargo fmt --all -- --check"
+    "rust:fmt": "cargo fmt --all -- --check",
+    "rust:clippy": "cargo clippy -- -D warnings"
```
Comment (matjazv) [inline package.json:35]: There are still 4 Clippy warnings in `smt.rs` and 6 warnings in `state_writer.rs`. In both files are Clippy warnings in tests.

In order to also check tests and non-default crate features we need to use:

```suggestion
    "rust:clippy": "cargo clippy --all-targets --all-features -- -D warnings"
```

**Review by matjazv — CHANGES_REQUESTED** [review]

File: package.json (hunk 3 above)
Comment (hrmhatef) [reply package.json]: Thank you for your good comments, it is done.

**Review by matjazv — APPROVED** [review]

**Review by shuse2 — APPROVED** [review]

**Pull request merged**